phylo/src/alignment/sequences.rs

```diff
@@ -0,0 +1,90 @@
+use crate::alphabets::{detect_alphabet, Alphabet, GAP};
+
+use bio::io::fasta::Record;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Sequences {
+    pub(crate) s: Vec<Record>,
+    pub(crate) aligned: bool,
+    pub(crate) msa_len: usize,
+    pub(crate) alphabet: Alphabet,
+}
+
+impl Sequences {
+    /// Creates a new Sequences object from a vector of bio::io::fasta::Record.
+    /// The Sequences object is considered aligned if all sequences have the same length.
+    pub fn new(s: Vec<Record>) -> Sequences {
+        let len = if s.is_empty() { 0 } else { s[0].seq().len() };
+        let alphabet = detect_alphabet(&s);
+        if s.iter().filter(|rec| rec.seq().len() != len).count() == 0 {
+            Sequences {
+                s,
+                aligned: true,
+                msa_len: len,
+                alphabet,
+            }
+        } else {
+            Sequences {
+                s,
+                aligned: false,
+                msa_len: 0,
+                alphabet,
+            }
+        }
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &Record> {
+        self.s.iter()
+    }
+
+    pub fn len(&self) -> usize {
+        self.s.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.s.is_empty()
+    }
+
+    pub fn get(&self, idx: usize) -> &Record {
+        &self.s[idx]
+    }
+
+    pub fn get_mut(&mut self, idx: usize) -> &mut Record {
+        &mut self.s[idx]
+    }
+
+    pub fn get_by_id(&self, id: &str) -> &Record {
```

Comment:
nit: this would be `from_id`

Reply:
This method is a getter, but `from_id()` by convention would be a constructor and would not take self. It is now renamed to `record_by_id()` as shown above.